api/grpcserver/v2beta1/malfeasance.go

```diff
@@ -145,6 +145,20 @@ func (s *MalfeasanceStreamService) Stream(
 	request *spacemeshv2beta1.MalfeasanceStreamRequest,
 	stream spacemeshv2beta1.MalfeasanceStreamService_StreamServer,
 ) error {
+	var sub *events.BufferedSubscription[events.EventMalfeasance]
+	if request.Watch {
+		matcher := malfeasanceMatcher{request}
+		var err error
+		sub, err = events.SubscribeMatched(matcher.match)
+		if err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
+		defer sub.Close()
+		if err := stream.SendHeader(metadata.MD{}); err != nil {
+			return err
+		}
+	}
```

Comment:
When it subscribes for "new proof" events before scanning the DB, there is a possibility of sending some proofs twice (when new are added while reading the DB). Previously there was a risk of missing some proofs. How about keeping a set of proofs sent from the DB? The code reading proofs from the subscription channel would need to see if the proof was already read from the DB (it doesn't need to update the set however anymore).

Reply:
That's why I originally changed the implementation from this to what it was before (this is basically a revert of some of the changes in #6557). A proper fix imo would be not to revert the code but instead fix the test that relies on reading the header from the stream for synchronization. wdyt?

Reply:
I reverted the implementation to what it was before and removed sending the header before streaming proofs and instead sleep in the test. I tried a few different approaches that were all more complex because `Stream` can be called from multiple clients concurrently and then I would need to track for which caller the handler already subscribed to events before sending them.

The main issue here is that `events` are global, if they were a service that is injected into the `MalfeasanceStreamService` then this synchronization wouldn't be needed. Since we will refactor events int he future eventually anyway, I think for now it's OK to instead just use a sleep in this test.